UnityProject/Assets/Scripts/Core/Input System/BlobMouseInputController.cs

```diff
@@ -0,0 +1,112 @@
+using Mirror;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Blob;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using Weapons;
+using Objects.Wallmounts;
+using Blob;
+
+/// <summary>
+/// Main entry point for handling all input events
+/// </summary>
+public class BlobMouseInputController : MouseInputController
+{
+	private BlobPlayer blobPlayer;
+
+	public bool placeOther;
+
+	public BlobConstructs blobConstructs;
+
+	public override void Start()
+	{
+		base.Start();
+		blobPlayer = GetComponent<BlobPlayer>();
+	}
+
+	public override void CheckMouseInput()
+	{
+		if (EventSystem.current.IsPointerOverGameObject())
+		{
+			//don't do any game world interactions if we are over the UI
+			return;
+		}
+
+		if (UIManager.IsMouseInteractionDisabled)
+		{
+			//still allow tooltips
+			CheckHover();
+			return;
+		}
+
+		if (CommonInput.GetMouseButtonDown(0))
+		{
+
+			if(ClicksFromBlobHud()) return;
+
+			//check ctrl+click for dragging
+			if (KeyboardInputManager.IsControlPressed())
+			{
+				//Place strong blob / reflective if strong blob already
+				blobPlayer.CmdTryPlaceStrongReflective(Camera.main.ScreenToWorldPoint(CommonInput.mousePosition).RoundToInt());
+				return;
+			}
+
+			if (KeyboardInputManager.IsShiftPressed())
+			{
+				//like above, send shift-click request, then do nothing else.
+				Inspect();
+				return;
+			}
+
+			if (KeyboardInputManager.IsAltPressed())
+			{
+				//Remove blob
+				blobPlayer.CmdRemoveBlob(Camera.main.ScreenToWorldPoint(CommonInput.mousePosition).RoundToInt());
+				return;
+			}
+
+			blobPlayer.CmdTryPlaceBlobOrAttack(Camera.main.ScreenToWorldPoint(CommonInput.mousePosition).RoundToInt());
+		}
+		else
+		{
+			CheckHover();
+		}
+	}
+
+	private bool ClicksFromBlobHud()
+	{
+		if (placeOther)
+		{
+			blobPlayer.CmdToggleRemove(true);
+
+			switch (blobConstructs)
```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6357029)